### PR TITLE
Network adapters can send and receive arbitrary message types

### PR DESCRIPTION
**Pull request by HerbCaudill**
Previously I added an AuthMessage type to automerge-repo:

https://github.com/automerge/automerge-repo/blob/830ffb6e57dbdd3c195bbd34fc1392165a44e027/packages/automerge-repo/src/network/messages.ts#L90-L102

We've redesigned the auth provider so that automerge-repo doesn't need to know it exists, so this doesn't belong here any more. But the auth provider still needs to be able to use the network adapter to send its own messages. So the network adapters need to be a bit more agnostic as to the type of messages they'll receive. Not completely agnostic -- we don't want messages to be of type `any` -- but they should take anything that looks like a message. 

This PR replaces the previous `Message` type, which was `RepoMessage | AuthMessage`, with the following: 

```ts
export type Message = {
  type: string

  /** The peer ID of the sender of this message */
  senderId: PeerId

  /** The peer ID of the recipient of this message */
  targetId: PeerId

  data?: Uint8Array

  documentId?: DocumentId
}```

### CODE REVIEW

**Comment by HerbCaudill** [comment]
Current dependencies on/for this PR:
* main
  * **PR #243** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/243?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #244** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/244?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #242** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/242?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #248** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/248?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #249** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/249?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #250** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/250?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/automerge/automerge-repo/250?utm_source=stack-comment).

**Comment by HerbCaudill** [comment]
Current dependencies on/for this PR:
* main
  * **PR #243** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/243?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #244** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/244?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #242** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/242?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #248** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/248?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #249** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/249?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #250** <a href="https://app.graphite.dev/github/pr/automerge/automerge-repo/250?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/automerge/automerge-repo/250?utm_source=stack-comment).

File: packages/automerge-repo/src/network/messages.ts [hunk 1]
```diff
@@ -128,38 +130,30 @@ export type RepoMessage =
...
-  documentId: DocumentId
-  syncState: SyncState
-}
-
 // TYPE GUARDS
 
-export const isValidRepoMessage = (message: Message): message is RepoMessage =>
-  typeof message === "object" &&
-  typeof message.type === "string" &&
-  typeof message.senderId === "string" &&
```
Comment (HerbCaudill) [inline packages/automerge-repo/src/network/messages.ts:156]: we don't need to assert stuff that typescript asserts for us

File: packages/automerge-repo/src/network/messages.ts [hunk 2]
```diff
@@ -21,102 +30,95 @@ export type SyncMessage = {
...
+export type RemoteSubscriptionControlMessage = {
+  type: "remote-subscription-change"
   senderId: PeerId
-
-  /** The peer ID of the recipient of this message */
   targetId: PeerId
 
-  /** The payload of the auth message (up to the specific auth provider) */
-  payload: TPayload
-}
```
Comment (HerbCaudill) [inline packages/automerge-repo/src/network/messages.ts:104]: automerge-repo doesn't know about auth

**Comment by pvh** [comment]
Rebased and merged.

**Pull request closed without merging**